dice-game-ui-angular/src/app/services/local-game-storage-service.ts

```diff
@@ -1,33 +1,70 @@
 import { Injectable } from '@angular/core';
 import { Guid } from 'guid-typescript';
-import { GameEngine, SimpleGameEngine, Size } from 'engine';
-import { GameStorageService } from './game-storage-service';
+import { GameEngine, SimpleGameEngine } from 'engine';
+import { GameStorageService, SavedGameEngine } from './game-storage-service';
 
 @Injectable()
 export class LocalGameStorageService implements GameStorageService {
-  createGame(gameEngine: GameEngine): Guid {
-    const guid = Guid.create();
-    localStorage.setItem(guid.toString(), gameEngine.fieldSize.width.toString());
-    return guid;
-  }
+  private readonly savedGamesKey = 'saved-games';
+
+  saveGame(gameEngine: GameEngine, guid: Guid = null): Guid {
+    const allSavedGames = this.getAllSavedGames();
+    if (!guid && allSavedGames.length === 0) {
+      throw new Error('This game was not created in the store.');
+    }
+
+    let savedGame: SavedGameEngine = allSavedGames.find(
+      (sg) => sg.guid === (guid ? guid.toString() : Guid.EMPTY)
+    );
+    // if saving previously saved game
+    if (guid) {
+      if (!savedGame) {
+        throw new Error('This game was not created in the store.');
+      }
+      savedGame.gameEngine = gameEngine;
+
+      // if saving a new game
+    } else {
+      if (savedGame) {
```

Comment:
Impossible condition. Saved game will never be found if you don't pass a guid to the function.

Reply:
Removed